units/suspend/suspend.pxu

```diff
@@ -2273,3 +2273,35 @@ estimated_duration: 0.5
 command:
  [ -e "${PLAINBOX_SESSION_SHARE}"/fwts_oops_results_after_s3.log ] && xz -c "${PLAINBOX_SESSION_SHARE}"/fwts_oops_results_after_s3.log
 _description: Attaches the FWTS oops results log to the submission after suspend
+
+id: suspend/system-suspend-to-ram
+after: suspend/suspend_advanced_auto
+category_id: com.canonical.plainbox::suspend
+_summary: System suspend to RAM check
+_description:
+ Check if system can suspend to RAM.
+ For more detail, plesae refer to https://www.kernel.org/doc/html/latest/admin-guide/pm/sleep-states.html.
+unit: job
+plugin: shell
+requires:
+ cpuinfo.platform in ('i386', 'x86_64')
+command:
+ echo "System suspended with \"$(dmesg | grep "PM: suspend entry" | tail -1 | sed 's/\[.*PM/PM/')\""
+ state=$(sudo dmesg | grep "PM: suspend entry" | tail -1 | cut -d'(' -f 2 | cut -d ')' -f 1)
+ echo "mem_sleep: $(cat /sys/power/mem_sleep)"
+ deep=$(cat /sys/power/mem_sleep | grep deep)
+ dmesg | grep ACPI | grep supports | sed 's/\[.*ACPI/ACPI/'
+ s3=$(dmesg | grep ACPI | grep supports | sed 's/\[.*ACPI/ACPI/' | grep S3)
+ if [ "$deep" != "" ] && [ "$s3" != "" ]; then
+     echo "The system supports suspend to RAM"
+     case "$state" in
+         "s2idle") echo "But, suspended to Suspend-to-Idle state" ;;
+         "shallow") echo "But, suspended to Standby state" ;;
```

Comment:
We also want to fail the test if the system has not entered suspend-to-ram, even if `/sus/power/mem_sleep` indicates it is supported.

I guess this is the situation we had with the B1400CEAE on earlier kernels: suspend-to-ram was supported by the hardware, but a kernel quirk set the system to suspend-to-idle by default because it failed to resume from suspend-to-ram. In those cases we want this test to fail, so we don't select models with that problem for PAYG deployments.

Reply:
Hmm ...  kind of lessons learned
ok